extensions/wikia/FirstContributions/api/ApiQueryFirstContributions.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/**
+ * API module which returns list of users' first edits, including revision id, date and user name
+ * @ingroup API
+ */
+class ApiQueryFirstContributions extends ApiQueryBase {
+
+	public function __construct( ApiBase $query, $moduleName ) {
+		parent::__construct( $query, $moduleName, 'fc' );
+	}
+
+	public function execute() {
+		$params = $this->extractRequestParams();
+
+		$this->addTables( 'recentchanges' );
+
+		$this->addFields( 'rc_this_oldid as id' );
+		$this->addFields( 'rc_timestamp as timestamp' );
+		$this->addFields( 'rc_user_text as author' );
+
+		$this->addWhere( 'rc_type < ' . RC_MOVE );
+		$this->addWhere( [ 'rc_bot' => 0 ] );
+
+		$this->addTimestampWhereRange( 'rc_timestamp', $params['dir'], $params['after'], $params['before'] );
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "before" 5 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "after" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)